definitions/artifacts/api.json

```diff
@@ -45,18 +45,39 @@
     },
     "specs": {
       "type": "object",
-      "additionalProperties": false,
-      "required": [
-        "api"
-      ],
-      "properties": {
-        "api": {
-          "version": {
-            "type": "string",
-            "description": "API Version"
+      "oneOf": [
```

Comment:
We currently require the api so we can show the version in the UI ... does this service not have a version?

We can also model this a little simpler (given we keep API required)

```js
"specs": {
      "type": "object",
      "additionalProperties": false, // we don't want people writing data that will never be used from massdriver_artifact... a typo could make it really hard to debug connections
      "required": [
        "api"
      ],
      "properties": {
        "api": {
          "version": {
            "type": "string",
            "description": "API Version"
          }
        },
        "aws": { // these are all optional, no need to do oneOf
          "$ref": "../specs/aws.json"
        },
        "gcp": {
          "$ref": "../specs/gcp.json"
        },
        "azure": {
          "$ref": "../specs/azure.json"
        }
      }
    }
```

Reply:
https://www.jsonschemavalidator.net/s/d9Go90gx

Reply:
I also don't see it in the resource's JSON:
![image](https://user-images.githubusercontent.com/8037512/213314446-24f5ad21-3550-4429-a713-5f487ecece20.png)

Reply:
In the comm service i'd set the `specs.api.version` to `n/a` in case they expose it in the future. Lets us keep the validation we do elsewhere (and stay consistent w/ other artdefs) and gives us fwd compat.

Reply:
Sounds good, will do!